setup/helpers/abs_helper.rb

```diff
@@ -654,24 +660,143 @@ def valid_abs_resource_hosts?(abs_resource_hosts)
-        else
-          puts "The specified resource host array is not valid: #{abs_resource_hosts}"
-          puts
-        end
-      rescue StandardError
-        # TODO: raise?
-        puts "JSON::ParserError encountered parsing the hosts array: #{abs_resource_hosts}"
+      # begin
+      #   hosts = JSON.parse(abs_resource_hosts)
+      #   host = hosts[0]
+      #   hostname = host["hostname"]
+      #   if !hostname.nil? && !hostname.empty?
+      #     is_valid = true
+      #   else
+      #     puts "The specified resource host array is not valid: #{abs_resource_hosts}"
+      #     puts
+      #   end
+      # rescue StandardError
+      #   # TODO: raise?
+      #   puts "JSON::ParserError encountered parsing the hosts array: #{abs_resource_hosts}"
+      # end
+
+      hosts = parse_abs_resource_hosts(abs_resource_hosts)
+      host = hosts[0]
+      hostname = host["hostname"]
+      if !hostname.nil? && !hostname.empty?
+        is_valid = true
+      else
+        puts "The specified resource host array is not valid: #{abs_resource_hosts}"
+        puts
       end
 
     end
 
     is_valid
   end
   # rubocop:enable Naming/AccessorMethodName
+
+  # Parses the specified abs_resource_hosts JSON
+  # note: consolidating this here since it is used by multiple methods
+  #
+  # @author Bill Claytor
+  #
+  # @param [JSON] abs_resource_hosts The hosts JSON to parse
+  #
+  # @raise [RuntimeError] If parsing is not successful
+  #
+  # @return [Array] The hosts array
+  #
+  # @example
+  #   hosts = parse_abs_resource_hosts(abs_resource_hosts)
+  #
+  def parse_abs_resource_hosts(abs_resource_hosts)
+    begin
+      hosts = JSON.parse(abs_resource_hosts)
+
+    # TODO: JSON::ParserError?
+    rescue StandardError => e
+      # TODO: raise? (current expected behavior is puts)
+      puts "#{e.inspect} encountered parsing the hosts array: #{abs_resource_hosts}"
+      hosts = nil
+    end
+    hosts
+  end
+
+  # Provisions a host with a tag for the specified role
+  # with the optionally specified size and volume_size
+  #
+  # @author Bill Claytor
+  #
+  # @param [String] role The value to specify for the role tag
+  # @param [String] size The EC2 instance size to use
+  # @param [String] volume_size The volume size to use
+  #
+  # @raise [RuntimeError] If provisioning is not successful
+  #
+  # @return [String] The hostname of the provisioned host
+  #
+  # @example
+  #   hostname = provision_host_for_role(role)
+  #
+  #   or
+  #
+  #   hostname = provision_host_for_role(role, "c5.xlarge", "40")
+  #
+  # TODO: spec test(s)
+  def provision_host_for_role(role, size = AWS_DEFAULT_SIZE, volume_size = AWS_VOLUME_SIZE)
+    host_to_provision = get_host_to_provision(role, size, volume_size)
+    abs_resource_hosts = get_abs_resource_hosts(host_to_provision)
+    raise "Unable to provision host via ABS" unless abs_resource_hosts
+
+    hosts = parse_abs_resource_hosts(abs_resource_hosts)
+    hostname = hosts[0]["hostname"]
```

Comment:
What if hosts is nil?

Reply:
I've updated `parse_abs_resource_hosts` to raise the error, and `valid_abs_resource_hosts?` rescues.